### ISSUE TITLE
Dryad API output change forces complete reingest of all Dryad material

### ISSUE THREAD

**Issue by plesubc** [ubc-library-rc/dryad2dataverse#3]
With version v0.7.09 of Dryad:

1. The Dryad  API has changed slightly
2. Dryad file download paths have changed, whether as a result of the new release or whether the internal structure of Dryad has changed is not clear.

This resulted in two things:

1. The output JSON of Dryad studies changed, which means *all* metadata was listed as changed as metadata changes were tracked by the changes in the JSON itself.

One notable feature: The "relationship" keywords appear to have changed. "article" is now "primary_article"

2. The file path was used in the identification of files. This means dryad2dataverse thought *all* files had changed.
As md5s are not generated by by Dryad, files were identified as unchanged when the file download path and the size were identical. File download paths have changed as they are apparently based on an internal database identifier, so dryad2dataverse reports that *all* files have changed.

Obviously, if all the files and all the metadata changed, that is a complete reharvest, which is by no means ideal.

Note that dryad2dataverse will still function as designed, but, in essence, there will be a new "version" of data which has basically the same metadata and data, and thus storage space will be doubled. Depending on the size of your dryad material, this may or may not be an issue.

Should you wish to see if an fix is possible and you have not already published any changes, if you have not published any updates since 21 October 2021, 

1.  Keep the October or earlier database as a reference
2.  Delete the unpublished drafts
3. Await a fix (which currently has no estimated time).

If you a have already updated and published, no action should be required and your current version of dryadd.py will function properly. You will just have duplicate files and very similar metadata.

**Comment by eugene-barsky** [comment]
"This means dryad2dataverse thought all files had changed." sigh...

**Comment by plesubc** [comment]
As of 0.7.06  2021-10-26 End of sprint release <https://github.com/CDL-Dryad/dryad-app/releases>:

When the API returns a “version” object, that object now includes a versionChanges field indicating whether data files were changed.

From <https://github.com/CDL-Dryad/dryad-app/blob/d68378d1fa0fffdd75f811de6eb2c63233251f82/stash/stash_api/public/openapi.yml>

```
 versionChanges:
              type: string
              description: Whether this version has had files_changed, or only metadata_changed
lastModificationDate:
              type: string
```
This will clearly help somewhat in identifying types of changes.

**Comment by plesubc** [comment]
From <https://github.com/CDL-Dryad/dryad-app/blob/a8ec4923fa1f39a222d0b0085d698c51a71377bb/stash/stash_api/app/models/stash_api/version/metadata.rb> the only possible values of versionChanges are `files_changed` and `metadata_changed`.

My examination of the json output shows that the fileIDs have in fact changed for all files. The mechanism used to identify was a change in any of fileID, size and name. If file_id is changeable. FileID was important because it would indicate a new upload if changed (or that was my assumption). Mismatches or even matches between just size and name are not necessarily reliable indicators of change. Maybe they will have to do.

**Comment by plesubc** [comment]
Resolved with v0.3.1 release
